config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java

```diff
@@ -90,9 +90,6 @@ private void ensureUnique(DeclaredZone zone, Set<DeclaredZone> zones) {
 
     /** Adds missing required steps and reorders steps to a permissible order */
     private static List<Step> completeSteps(List<Step> steps) {
-        // Ensure no duplicate deployments to the same zone
-        steps = new ArrayList<>(new LinkedHashSet<>(steps));
-        
```

Comment:
This no longer works, after parallell steps. It is also superseded by the validateSteps which is called after this method in the constructor.